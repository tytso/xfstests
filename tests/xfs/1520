#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-later
# Copyright (c) 2020 Oracle, Inc.  All rights reserved.
#
# FS QA Test No. 1520
#
# Populate a XFS filesystem and fuzz every leaf1-format dir block field.
# Try online repair and, if necessary, offline repair,
# to test the most likely usage pattern.

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 7 15

_cleanup()
{
	cd /
	rm -rf $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/populate
. ./common/fuzzy

# real QA test starts here
_supported_fs xfs
_require_scratch_xfs_fuzz_fields
_disable_dmesg_check

echo "Format and populate"
_scratch_populate_cached nofill > $seqres.full 2>&1

echo "Find leaf1-format dir block"
_scratch_mount
inum=$(stat -c '%i' $SCRATCH_MNT/S_IFDIR.FMT_LEAF)
blk_sz=$(_get_block_size $SCRATCH_MNT)
_scratch_unmount

leaf_offset=$(( (2 ** 35) / blk_sz))
echo "Fuzz leaf1-format dir block"
_scratch_xfs_fuzz_metadata '' 'both'  "inode ${inum}" "dblock ${leaf_offset}" >> $seqres.full
echo "Done fuzzing leaf1-format dir block"

# success, all done
status=0
exit
