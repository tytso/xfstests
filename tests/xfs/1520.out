QA output created by 1520
Format and populate
Find leaf1-format dir block
Fuzz leaf1-format dir block
Done fuzzing leaf1-format dir block
